aptos-move/framework/aptos-framework/sources/configs/stake.move

```diff
@@ -1013,34 +1009,37 @@ module aptos_framework::stake {
         pool_address: address,
         validator_set_config: &ValidatorSetConfiguration,
     ) acquires StakePool, StakePoolEvents, AptosCoinCapabilities, ValidatorConfig {
+        let stake_pool = borrow_global_mut<StakePool>(pool_address);
         let validator_config = borrow_global<ValidatorConfig>(pool_address);
         let num_missed_votes = *vector::borrow(&validator_perf.missed_votes, validator_config.validator_index);
         let num_blocks = validator_perf.num_blocks;
         let num_successful_votes = num_blocks - num_missed_votes;
 
-        let current_time = timestamp::now_seconds();
-        // Remaining lockup time needs to be at least one so the rewards calculation still works.
-        let remaining_lockup_time = 1;
-        let stake_pool = borrow_global_mut<StakePool>(pool_address);
-        if (current_time < stake_pool.locked_until_secs) {
-            remaining_lockup_time = stake_pool.locked_until_secs - current_time;
-        };
-
-        // Always merge rewards to active. If a validator's remaining lockup has already expired and they're withdrawing
-        // coins, their rewards would be 0 anyway as remaining lockup = 0.
         let rewards = mint_reward(
             validator.voting_power,
             num_blocks,
             num_successful_votes,
-            remaining_lockup_time,
             validator_set_config,
         );
         let rewards_amount = coin::value<AptosCoin>(&rewards);
-        coin::merge(&mut stake_pool.active, rewards);
+        // Merge rewards into active unless the validator is unlocking all of their stake.
```

Comment:
Minting to both active and pending_inactive is also an option but it can add complexity, especially if the pending_inactive amount is very small (which can round rewards for it down to 0)